src/main/java/org/folio/services/CallNumberUtils.java

```diff
@@ -0,0 +1,18 @@
+package org.folio.services;
+
+import org.marc4j.callnum.CallNumber;
+import org.marc4j.callnum.LCCallNumber;
+
+import java.util.Optional;
+
+public class CallNumberUtils {
+
+  private CallNumberUtils() {}
+
+  public static Optional<String> getShelfKeyFromCallNumber(String callNumberParam) {
+    if (callNumberParam == null) return Optional.empty();
+    CallNumber callNumber = new LCCallNumber();
+    callNumber.parse(callNumberParam);
+    return  callNumber.isValid() ? Optional.of(callNumber.getShelfKey()) : Optional.empty();
+  }
+}
```

Comment:
Why is this separate to the effective call number components utility class?  Is the intention to have a class specific to generation to the shelving order?

Reply:
At this time marc4j library implemented Library Of Congress Call Number Type. If you see at CallNumberType attribute of Item you could found that it has a bunch of predefined values. It means that we can meet with unpredictable development implementation of handlers for the other types of Call Number. As for me good idea to define separate class where any implementation could be integrated without affect any classes on more high level of logic.

Reply:
I think what you are describing is that this class is intended to hide the use of the external marc4j library from the rest of the system. Have I understood that correctly?

Reply:
Sure. Any other extention or library can be added without any changes in source code placed higer than library wrapper.